nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -38,7 +38,9 @@ let
 
     configurationLimit = if cfg.configurationLimit == null then 0 else cfg.configurationLimit;
 
-    inherit (cfg) consoleMode graceful editor;
+    inherit (cfg) consoleMode graceful editor efiBootOptionDescription;
+
+    setEfiBootOptionDescription = cfg.efiBootOptionDescription != null;
```

Comment:
Why not do this test on the python side?

Reply:
I wasn't sure of how to do that reliably. If `efiBootOptionDescription` is `null`, the substitution logic will not change the value. I could recreate the "@efiBootOptionDescription@" through string concatenation to do the comparison. It seemed a little odd so I did it this way.